traversal/selector/exploreRecursive.go

```diff
@@ -34,7 +34,8 @@ import (
 type ExploreRecursive struct {
 	sequence Selector // selector for element we're interested in
 	current  Selector // selector to apply to the current node
-	maxDepth int
+	depth    int
+	hasDepth bool
```

Comment:
This does all look about right, but I might consider extracting one more type here in the go code to stay topologically closer to the schema, and embed it here... like:

```
type RecursionLimit_Mode uint8

const (
    RecursionLimit_None RecursionLimit_Mode = 0
    RecursionLimit_Depth RecursionLimit_Mode = 1
)

type RecursionLimit struct {
    mode RecursionLimit_Mode
    depth int
}
```

... and then embed that, and most of the rest of the code stays the same, except a bunch of `if` blocks change to `switch` (which seems more correct if we do ever imagine adding more members to the limit union).

(Admittedly, it's always kind of frustrating to decide which of the 14(?) different kinds of sort-of-acts-like-a-union non-native construct in Go to use to implement a conceptual union.  I picked one at semirandom here.)